### PR TITLE
Produce bundles with Rollup

### LINKED ISSUE #41: Produce bundles
Use rollup ? Or webpack ?

When a new tag is created :
- [ ] Build a dist and commit it in dist repository
- [ ] Attach it to the associated github release
- [ ] Deliver the files with https://www.jsdelivr.com/

### PR DESCRIPTION
**Pull request by charlyx**
fix #41

### CODE REVIEW

**Comment by nlepage** [comment]
I moved this to 1.0, what's the status, did you make it work with rollup ?

**Comment by nlepage** [comment]
Tested in Chrome with following page :
```html
<html>
<head>
<script src="immutadot.js"></script>
<script src="https://cdn.jsdelivr.net/g/lodash@4(lodash.min.js+lodash.fp.min.js)"></script>
<script>
(function() {
  var fp = _.noConflict()
  _.fp = fp
})()
</script>
<script src="immutadot-lodash.js"></script>
<script>
console.log(immutadot.set({ english: { greeting: 'Hi' } }, 'english.greeting', 'Hello'))
console.log(immutadot.push({ i18n: { languages: ['English', 'French'] } }, 'i18n.languages', 'German', 'Spanish'))
const animals = {
  weasels: [
    {
      vernacularName: 'badger',
      scientificName: 'Meles meles'
    },
    {
      vernacularName: 'otter',
    }
  ]
}
console.log(immutadot_.capitalize(animals, 'weasels[:].vernacularName'))
</script>
</head>
<body></body>
</html>
```

**Comment by nlepage** [comment]
TODO:
 - [ ] Test on various browsers to validate they are supported (especially older ones)
 - [ ] Put back a minified version (ideally `yarn dist` should generate both minified and not minified)

**Comment by nlepage** [comment]
What do you think of the name `immutadot_` for immutadot-lodash, maybe it is unclear ?

File: config/rollup.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import commonjsPlugin from 'rollup-plugin-commonjs'
+import fs from 'fs'
+import { resolve } from 'path'
+import resolvePlugin from 'rollup-plugin-node-resolve'
+import uglify from 'rollup-plugin-uglify'
+
+const root = resolve(__dirname, '..')
+const nodeModules = resolve(root, 'node_modules')
+
+const makeBundle = (name, options = {}, minify = false) => {
```
Comment (frinyvonnick) [inline config/rollup.js]: I recommend to comment each statement because we won't remember why we have put some options in 6 month.

Comment (charlyx) [reply config/rollup.js]: I'll let @nlepage explain what he has done to make it work with immutadot-lodash :P

**Comment by nlepage** [comment]
I'm sorry I just saw that jsdelivr.net handle minifying files on demand, so it's not necessary to do it ourselves...

**Comment by nlepage** [comment]
Should we really use umd ? I don't think bundles need to support commonjs...

**Comment by charlyx** [comment]
@nlepage shall we change output format to iife then?

**Comment by nlepage** [comment]
Tested ✔️ on IE 11 🙃

**Pull request merged**